### PR TITLE
Adjusted firstRelatedAudioItem method to accomodate voice

### PR DESCRIPTION
**Pull request by skytan4**

### CODE REVIEW

**Comment by nickmshelley** [comment]
Wrong repo. Not sure why we have this fork. Use CrossWaterBridge/LDSContent instead.

File: LDSContent/ItemPackage+RelatedAudioItem.swift [hunk 1]
```diff
@@ -44,8 +44,8 @@ public extension ItemPackage {
         
     }
     
-    public func firstRelatedAudioItemForSubitemWithURI(subitemURI: String) -> RelatedAudioItem? {
-        return db.pluck(RelatedAudioItemTable.table.join(SubitemTable.table.filter(SubitemTable.uri == subitemURI), on: SubitemTable.table[SubitemTable.id] == RelatedAudioItemTable.table[RelatedAudioItemTable.subitemID]).order(SubitemTable.table[SubitemTable.id])).map { RelatedAudioItemTable.fromNamespacedRow($0) }
+    public func firstRelatedAudioItemForSubitemWithURI(subitemURI: String, relatedAudioVoice: RelatedAudioVoice) -> RelatedAudioItem? {
+        return db.pluck(RelatedAudioItemTable.table.join(SubitemTable.table.filter(SubitemTable.uri == subitemURI && (RelatedAudioItemTable.voice == relatedAudioVoice || RelatedAudioItemTable.voice == nil)), on: SubitemTable.table[SubitemTable.id] == RelatedAudioItemTable.table[RelatedAudioItemTable.subitemID]).order(SubitemTable.table[SubitemTable.id])).flatMap { RelatedAudioItemTable.fromNamespacedRow($0) }
```
Comment (stephanheilner) [inline LDSContent/ItemPackage+RelatedAudioItem.swift]:
```swift
var query = RelatedAudioItemTable.table.join(SubitemTable.table.filter(SubitemTable.uri == subitemURI), on: SubitemTable.table[SubitemTable.id] == RelatedAudioItemTable.table[RelatedAudioItemTable.subitemID]).order(SubitemTable.table[SubitemTable.id])

if let relatedAudioVoice = relatedAudioVoice {
    query = query.filter(RelatedAudioItemTable.voice == relatedAudioVoice)
}

return db.pluck(query).map { RelatedAudioItemTable.fromNamespacedRow($0) }
```

File: LDSContent/ItemPackage+RelatedAudioItem.swift [hunk 2]
```diff
@@ -44,8 +44,8 @@ public extension ItemPackage {
         
     }
     
-    public func firstRelatedAudioItemForSubitemWithURI(subitemURI: String) -> RelatedAudioItem? {
-        return db.pluck(RelatedAudioItemTable.table.join(SubitemTable.table.filter(SubitemTable.uri == subitemURI), on: SubitemTable.table[SubitemTable.id] == RelatedAudioItemTable.table[RelatedAudioItemTable.subitemID]).order(SubitemTable.table[SubitemTable.id])).map { RelatedAudioItemTable.fromNamespacedRow($0) }
+    public func firstRelatedAudioItemForSubitemWithURI(subitemURI: String, relatedAudioVoice: RelatedAudioVoice) -> RelatedAudioItem? {
```
Comment (stephanheilner) [inline LDSContent/ItemPackage+RelatedAudioItem.swift]: public func firstRelatedAudioItemForSubitemWithURI(subitemURI: String, relatedAudioVoice: RelatedAudioVoice? = nil) -> RelatedAudioItem?

**Comment by stephanheilner** [comment]
Ok, I did me review here though, so make these changes when you re-do this in the right place.

File: LDSContent/ItemPackage+RelatedAudioItem.swift (hunk 1 above)
Comment (skytan4) [inline LDSContent/ItemPackage+RelatedAudioItem.swift]: That doesn't work, I'll have to see if I can adjust it to get it to work. Are you wanting it like this for readability?

Comment (skytan4) [inline LDSContent/ItemPackage+RelatedAudioItem.swift]: Never mind, duh content does not have voice yet, thats why... Sorry

**Pull request closed without merging**